src/main/angular/app/graph/graph.component.html

```diff
@@ -22,7 +22,9 @@ <h1>Nothing to display! <br />
 </div>
 
 <div id="show-chart" *ngIf="shouldShowChart">
-  <div id="date-container" [style.width.px]="properties.width">
+  <div id="top-container" [style.width.px]="properties.width * 3 / 4">
+    <mat-spinner *ngIf="activeRequest" diameter="40"></mat-spinner>
```

Comment:
Did you consider a progress bar instead? Without a screenshot, it's hard to tell what the better User Experience would be. 

Link to progress bar.
https://material.angular.io/components/progress-bar/overview

Reply:
Yes. To my eye, the spinner fits in better with the rest of the page. 
I think it better represents the query in progress than something like the [query progress bar](https://material.angular.io/components/progress-bar/overview#query).